pkg/operator/event.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Some of the code below came from https://github.com/coreos/etcd-operator
+which also has the apache 2.0 license.
+*/
+package operator
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"k8s.io/client-go/pkg/api/unversioned"
+	kwatch "k8s.io/client-go/pkg/watch"
+)
+
+type Event struct {
+	Type   kwatch.EventType
+	Object *Cluster
+}
+
+type rawEvent struct {
+	Type   kwatch.EventType
+	Object json.RawMessage
+}
+
+func pollEvent(decoder *json.Decoder) (*Event, *unversioned.Status, error) {
```

Comment:
Kubernetes already maintains a util class that does pooling, wait etc. We should use this instead rather than creating and maintaining our own. https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go